cuegui/cuegui/MenuActions.py

```diff
@@ -907,6 +907,34 @@ def stagger(self, rpcObjects=None):
         self.cuebotCall(__layer.staggerFrames, "Stagger Frames Failed",
                         frameRange, int(increment))
 
+    previewMain_info = ["Preview Main", None, "image"]
+
+    # pylint: disable=broad-except
+    def previewMain(self, rpcObjects=None):
+        try:
+            layer = self._getOnlyLayerObjects(rpcObjects)[0]
+            if layer is not None:
+                outputs = layer.getOutputPaths()
+                if len(outputs) > 0:
+                    cuegui.Utils.previewOutputs([outputs[0]])
+        except Exception as e:
+            QtWidgets.QMessageBox.critical(None, "Preview Error",
+                                           "Error displaying preview frames, %s" % e)
+
+    previewAll_info = ["Preview All", None, "images"]
+
+    # pylint: disable=broad-except
+    def previewAll(self, rpcObjects=None):
+        try:
+            layer = self._getOnlyLayerObjects(rpcObjects)[0]
+            if layer is not None:
+                outputs = layer.getOutputPaths()
+                if len(outputs) > 0:
+                    cuegui.Utils.previewOutputs(outputs)
```

Comment:
Same suggestion of the previous comment

Reply:
I this case though outputs could actually be an empty list and shouldn't trigger the preview command. (it would just open the app without any files)

Reply:
A better way is to gray out the preview command if no outputs are present. In this case (since it's for layers and not frames) it would work, however for frames, if outputs are not present, it would fall back to using the Katana method.

Reply:
@DiegoTavares any thoughts on this?